DataFormats/PatCandidates/src/PackedCandidate.cc

```diff
@@ -62,9 +62,12 @@ void pat::PackedCandidate::packVtx(bool unpackAfterwards) {
 }
 
 void pat::PackedCandidate::unpack() const {
-    p4_ = PolarLorentzVector(MiniFloatConverter::float16to32(packedPt_),
+    float pt = MiniFloatConverter::float16to32(packedPt_);
+    double shift = (pt<1. ? 0.1*pt : 0.1/pt); // shift particle phi to break degeneracies in angular separations
+    double phi = int16_t(packedPhi_)*3.2f/std::numeric_limits<int16_t>::max() + shift*3.2/std::numeric_limits<int16_t>::max();
```

Comment:
Since particles are placed on an eta-phi grid, the degeneracy occurs in dR separation of particle pairs. In other words, there can be more than one pair with the same dR separation and this is particularly troublesome for the CA algorithm which uses dR as the distance measure (anti-kt and kt break the degenracy by also using particle pt). These degeneracies mostly affect early stages of jet clustering so essentially jets of any size are affected, not just fat jets with large cone size.

The ghost association will still work just fine since the particle pt's are not changed (0.1*pt, for pt<1, only determines by how much to shift the phi value in units of the grid spacing).